tests/test_templates_state_preparations.py

```diff
@@ -0,0 +1,218 @@
+        ([1, 0], [0, 2], [1, 0, 0]),
+        ([1, 1, 0], [0, 1, 2], [1, 1, 0]),
+        ([1, 0, 1], [0, 1, 2], [1, 0, 1]),
+    ])
+    # fmt: on
+    def test_state_preparation(self, tol, qubit_device_3_wires, basis_state, wires, target_state):
+        """Tests that the template BasisStatePreparation integrates correctly with PennyLane."""
+
+        @qml.qnode(qubit_device_3_wires)
+        def circuit():
+            BasisStatePreparation(basis_state, wires)
+
+            # Pauli Z gates identify the basis state
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2))
+
+        # Convert from Pauli Z eigenvalues to basis state
+        output_state = [0 if x == 1.0 else 1 for x in circuit()]
+
+        assert np.allclose(output_state, target_state, atol=tol, rtol=0)
+
+    # fmt: off
+    @pytest.mark.parametrize("basis_state,wires,error_message", [
+        ([0], [0, 1], "Number of qubits must be equal to the number of wires"),
+        ([0, 1], [0], "Number of qubits must be equal to the number of wires"),
+        ([0], 0, "Wires must be passed as a list of integers"),
+        ([3], [0], "Basis state must only consist of 0s and 1s"),
+        ([1, 0, 2], [0, 1, 2], "Basis state must only consist of 0s and 1s"),
+    ])
+    # fmt: on
+    def test_errors(self, basis_state, wires, error_message):
+        """Tests that the correct error messages are raised."""
+
+        with pytest.raises(ValueError, match=error_message):
+            BasisStatePreparation(basis_state, wires)
+
+    class TestMottonenStatePreparation:
+        """Tests the template MottonenStatePreparation."""
+
+        # def test_basis_state_mapping(self, num_wires):
+        #     """Tests that MottonenStatePreparation maps basis states to basis states."""
+        #     eye = np.eye(num_wires**2)
+        #     dev = qml.device("default.qubit", wires=num_wires)
+
+        #     idx_maps = []
+        #     bin_maps = []
+
+        #     for row in eye:
+        #         dev.reset()
+
+        #         @qml.qnode(dev)
+        #         def circuit():
+        #             MottonenStatePreparation(row, [0, 1, 2, 3])
+
+        #             return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2)), qml.expval(qml.PauliZ(3))
+
+        #         circuit()
+
+        #         state = dev._state
+
+        #         assert
+
+        #         print("Circuit maps\n    {}\n -> {}".format(row, np.real(np.round(state, 2))))
+        #         idx_maps.append((np.argmax(row), np.argmax(np.real(np.round(state, 2)))))
+        #         bin_maps.append(("{0:04b}".format(np.argmax(row)), "{0:04b}".format(np.argmax(np.real(np.round(state, 2))))))
+
+        #     #print(idx_maps)
+        #     #print(bin_maps)
+        #     for i in range(16):
+        #         print("{} -> {}, {} -> {}".format(idx_maps[i][0], idx_maps[i][1], bin_maps[i][0], bin_maps[i][1]))
+        #     raise Exception("X")
```

Comment:
How come this test is commented out?

Reply:
It was the test I used to get the code working, I thought I might still need it